build.sbt

```diff
@@ -61,17 +65,23 @@ val sharedSettings = Project.defaultSettings ++ scalariformSettings ++  Seq(
   scalacOptions ++= Seq(
     "-unchecked",
     "-deprecation",
-    "-optimize",
     "-Xlint",
     "-language:implicitConversions",
     "-language:higherKinds",
     "-language:existentials"),
 
-  scalacOptions <++= (scalaVersion) map { sv =>
-      if (sv startsWith "2.10")
-        Seq("-Xdivergence211")
-      else
-        Seq()
+  scalacOptions ++= {
+    if (scalaVersion.value startsWith "2.10")
+      Seq("-Xdivergence211")
+    else
+      Seq()
+  },
+
+  scalacOptions ++= {
+    if (scalaVersion.value startsWith "2.12")
```

Comment:
I'm not so sure we should be turning on optimization, actually. I have heard that this is best for end projects, not libraries, but I guess we can leave it as is for now.